include/ginkgo/core/base/polymorphic_object.hpp

```diff
@@ -57,6 +57,8 @@ class PolymorphicObject : public log::EnableLogging<PolymorphicObject> {
                                                                     this);
     }
 
+    virtual void validate_data() const {}
```

Comment:
```suggestion
    /**
     * Throws gko::InvalidData exception if the data inside the arrays is
     * invalid.
     */
    virtual void validate_data() const {}
```
you can move the description here.
Then you do not need the description when overriding it.